pkg/discovery/oci_dbbacked.go

```diff
@@ -30,6 +31,18 @@ type DBBackedOCIDiscovery struct {
 	// E.g., harbor.my-domain.local/tanzu-cli/plugins/plugins-inventory:latest
 	// This image contains a single SQLite database file.
 	image string
+	// criteria specified different conditions that a plugin must respect to be discovered.
+	// This allows to filter the list of plugins that will be returned.
+	criteria *PluginDiscoveryCriteria
+	// pluginDataDir is the location where the plugin data will be stored once
+	// extracted from the OCI image
+	pluginDataDir string
+	// inventory is the pluginInventory to be used by this discovery.
+	inventory plugininventory.PluginInventory
```

Comment:
The use of this field allows the unit tests to change the `inventory` to use a stub.